src/main/kotlin/com/team4099/robot2023/subsystems/elevator/Elevator.kt

```diff
@@ -0,0 +1,356 @@
+            Logger.recordOutput("Elevator/reverseLimitReached", reverseLimitReached)
+        }
+        var nextState = currentState
+        when (currentState) {
+            ElevatorState.UNINITIALIZED -> {
+                nextState = fromElevatorRequestToState(currentRequest)
+            }
+            ElevatorState.OPEN_LOOP -> {
+                setOutputVoltage(elevatorVoltageTarget)
+                nextState = fromElevatorRequestToState(currentRequest)
+            }
+            ElevatorState.TARGETING_POSITION -> {
+                if ((elevatorPositionTarget != lastRequestedPosition) || (elevatorVelocityTarget != lastRequestedVelocity)) {
+                    val preProfileGenerate = Clock.realTimestamp
+                    elevatorProfile = TrapezoidProfile(elevatorConstraints, TrapezoidProfile.State(elevatorPositionTarget, elevatorVelocityTarget), TrapezoidProfile.State(inputs.elevatorPosition, inputs.elevatorVelocity))
+                    val postProfileGenerate = Clock.realTimestamp
+                    Logger.recordOutput("/Elevator/profileGenerationMS", postProfileGenerate.inSeconds - preProfileGenerate.inSeconds)
+                    Logger.recordOutput("Elevator/initialPosition", elevatorProfile.initial.position.inInches)
+                    Logger.recordOutput("Elevator/initialVelocity", elevatorProfile.initial.velocity.inInchesPerSecond)
+                    timeProfileGeneratedAt = Clock.fpgaTime
+                    lastRequestedPosition = elevatorPositionTarget
+                    lastRequestedVelocity = elevatorVelocityTarget
+                }
+                val timeElapsed = Clock.fpgaTime - timeProfileGeneratedAt
+                val profilePosition = elevatorProfile.calculate(timeElapsed)
+                setPosition(profilePosition)
+                Logger.recordOutput("Elevator/completedMotionProfile", elevatorProfile.isFinished(timeElapsed))
+                Logger.recordOutput("Elevator/profileTargetVelocity", profilePosition.velocity.inInchesPerSecond)
+                Logger.recordOutput("Elevator/profileTargetPosition", profilePosition.position.inInches)
+                nextState = fromElevatorRequestToState(currentRequest)
+                if (! (currentState.equivalentToRequest(currentRequest))) {
+                    lastRequestedVelocity = -999.inches.perSecond
+                    lastRequestedPosition = -999.inches
+                }
+            }
+            ElevatorState.HOME -> {
+                if (inputs.leaderStatorCurrent < ElevatorConstants.HOMING_STATOR_CURRENT) {
+                    lastHomingStatorCurrentTripTime = Clock.fpgaTime
+                }
+                if (! inputs.isSimulating && (! isHomed && inputs.leaderStatorCurrent < ElevatorConstants.HOMING_STATOR_CURRENT && Clock.fpgaTime - lastHomingStatorCurrentTripTime < ElevatorConstants.HOMING_STALL_TIME_THRESHOLD)) {
+                    setHomeVoltage(ElevatorConstants.HOMING_APPLIED_VOLTAGE)
```

Comment:
i think you should straight up do `io.setVoltage()` here. reduce the clutter of the file bc home voltage shouldn't care about soft limits